assets/backport/github/github.go

```diff
@@ -0,0 +1,311 @@
+	cherryParent := cherryCommit.Parents[0]
+
+	// Temporarily set the parent of the branch HEAD to the parent of the commit
+	// to cherry-pick so they are siblings.
+	err := c.createSiblingCommit(ctx, branchName, headBranchCommit, cherryParent)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	// When git performs the merge, it detects that the parent of the branch commit that is
+	// being merged onto matches the parent of the cherry pick commit, and merges a tree of size 1.
+	// The merge commit will contain the delta between the file tree in target branch and the
+	// commit to cherry-pick.
+	merge, err := c.merge(ctx, branchName, cherryCommit.GetSHA())
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	mergeTree := merge.GetTree()
+
+	updatedCommit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		headBranchCommit.GetSHA())
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	// Create the actual cherry-pick commit on the target branch containing the merge commit tree.
+	commit, _, err := c.Client.Git.CreateCommit(ctx, c.Organization, c.Repository, &go_github.Commit{
+		Message: cherryCommit.Message,
+		Tree:    mergeTree,
+		Parents: []*go_github.Commit{
+			updatedCommit,
+		},
+	})
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	// Overwrite the merge commit and its parent on the branch by the newly created commit.
+	// The result will be equivalent to what would have happened with a fast-forward merge.
+	sha := commit.GetSHA()
+	refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+	_, _, err = c.Client.Git.UpdateRef(ctx, c.Organization, c.Repository, &go_github.Reference{
+		Ref: go_github.String(refName),
+		Object: &go_github.GitObject{
+			SHA: go_github.String(sha),
+		},
+	}, true)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	return mergeTree, sha, nil
+}
+
+// createSiblingCommit creates a commit with the passed in commit's tree and parent
+// and updates the passed in branch to point at that commit.
+func (c *Client) createSiblingCommit(ctx context.Context, branchName string, branchHeadCommit *go_github.Commit, cherryParent *go_github.Commit) error {
+	tree := branchHeadCommit.GetTree()
+
+	// This will be the temp commit, commit is lost.
+	commit, _, err := c.Client.Git.CreateCommit(ctx, c.Organization, c.Repository, &go_github.Commit{
+		Message: go_github.String("field-not-required"),
+		Tree:    tree,
+		Parents: []*go_github.Commit{
+			cherryParent,
+		},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	sha := commit.GetSHA()
+
+	refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+	_, _, err = c.Client.Git.UpdateRef(ctx, c.Organization, c.Repository, &go_github.Reference{
+		Ref: go_github.String(refName),
+		Object: &go_github.GitObject{
+			SHA: go_github.String(sha),
+		},
+	}, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// createBranchFrom creates a branch from the passed in branch's HEAD.
```

Comment:
```suggestion
// createBranchFrom creates a new branch pointing at the same commit as the supplied branch.
```
Is that what this is doing?